packages/inertia/src/inertia.js

```diff
@@ -278,30 +337,32 @@ export default {
   },
 
   remember(data, key = 'default') {
-    let newState = { ...window.history.state }
-    newState.cache = newState.cache || {}
-    newState.cache[key] = data
-    this.replaceState(newState)
+    let page = { ...this.page }
+    page.rememberedState[key] = data
+    this.replaceState(page)
   },
 
   restore(key = 'default') {
-    if (window.history.state.cache && window.history.state.cache[key]) {
-      return window.history.state.cache[key]
+    if (window.history.state.rememberedState && window.history.state.rememberedState[key]) {
+      return window.history.state.rememberedState[key]
     }
```

Comment:
```suggestion
    return window.history.state?.rememberedState?.[key]
```

Reply:
This is awesome. 😍

Reply:
`?->` soon in php (8)! 😉

Reply:
Honestly the feature I'm looking forward to the most in PHP 8! 🙌